oauth2_http/java/com/google/auth/http/HttpCredentialsAdapter.java

```diff
@@ -44,16 +55,58 @@ public void initialize(HttpRequest request) throws IOException {
     }
     for (Map.Entry<String, List<String>> entry : credentialHeaders.entrySet()) {
       String headerName = entry.getKey();
-      List<String> requestValues = getHeadersValue(requestHeaders, headerName);
-      if (requestValues == null) {
-        requestValues = new ArrayList<String>();
-        requestHeaders.put(headerName, requestValues);
+      List<String> requestValues = new ArrayList<String>();
```

Comment:
It is related because on retry, initialize is called again to reset the Authorization value.